app/ante/eth.go

```diff
@@ -549,3 +549,25 @@ func (issd EthIncrementSenderSequenceDecorator) AnteHandle(ctx sdk.Context, tx s
 	// set the original gas meter
 	return next(ctx, tx, simulate)
 }
+
+// NOTICE: Adapted from ValidateBasicDecorator from cosmos-sdk, ignore ErrNoSignatures
+type EthValidateBasicDecorator struct{}
+
+func NewEthValidateBasicDecorator() EthValidateBasicDecorator {
+	return EthValidateBasicDecorator{}
+}
+
+func (vbd EthValidateBasicDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// no need to validate basic on recheck tx, call next antehandler
+	if ctx.IsReCheckTx() {
+		return next(ctx, tx, simulate)
+	}
+
```

Comment:
Additionally we could also check that number of msg = 1

https://github.com/tharsis/ethermint/blob/main/app/ante/eth.go#L54

I think we can delete this check in `EthSigVerificationDecorator` and put it here instead 

```
	if len(tx.GetMsgs()) != 1 {		return ctx, stacktrace.Propagate(			sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "only 1 ethereum msg supported per tx, got %d", len(tx.GetMsgs())),			"",		)	}
	
	```

Reply:
https://github.com/tharsis/ethermint/blob/main/app/ante/eth.go#L76
seemingly `EthSigVerificationDecorator` is the only handler that check and rely on this assumption, maybe other members can comment on this?

Reply:
as long as it's checked on the first one, it's ok. We could support multiple msgs per tx but that would be a separate improvement